src/sietch/distributed_dune.ml

```diff
@@ -35,10 +40,39 @@ let connect t uri =
     |> async_ok
   in
   let* client =
+    let implementations =
+      let get () h =
+        let hash = hash h in
+        let path = Local.file_path t.cache hash |> Path.to_string in
+        let f () =
+          let ( let+ ) = Async.( >>| ) in
+          let ( and+ ) = Async.Deferred.both in
+          let+ stats = Async.Unix.stat path
+          and+ contents = Async.Reader.open_file path in
+          Async.Pipe.concat
+            [ Async.Pipe.of_list
+                [ Core.Either.second (stats.perm land 0o100 <> 0) ]
+            ; Async.Pipe.map ~f:Core.Either.first (Async.Reader.pipe contents)
+            ]
+        in
+        Async.Deferred.map (Async.try_with f)
+          ~f:(Result.map_error ~f:(fun _ -> ()))
+      in
+      match
+        Async.Rpc.Implementations.create
+          ~implementations:
+            [ Async.Rpc.Pipe_rpc.implement
+                Dune_distributed_storage.Rpc.block_get get
+            ]
+          ~on_unknown_rpc:`Raise
+      with
+      | Result.Ok impls -> impls
+      | Result.Error _ -> failwith "duplicate implementations"
```

Comment:
Seems worth specifying the type of the ignored value, as otherwise I had to double-check that we're not dropping any other types of exceptions on the floor. (Indeed, we don't!)